File: python/integracion_con_bases_de_datos/actualizar_contacto/actualizar_contacto.py
from conexion import conectar
def guardar_contacto(nombre, telefono, email):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = "INSERT INTO contactos (nombre, telefono, email) VALUES (%s, %s, %s)"
        cursor.execute(sql, (nombre, telefono, email))
        conexion.commit()
        cursor.close()
        conexion.close()
        print("✅ Contacto guardado")
def mostrar_contactos():
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        cursor.execute("SELECT * FROM contactos")
        resultados = cursor.fetchall()
        if resultados:
            print("\n📋 Contactos:\n")
            for contacto in resultados:
                print(f"ID: {contacto[0]} | Nombre: {contacto[1]} | Tel: {contacto[2]} | Email: {contacto[3]}")
        else:
            print("⚠️ No hay contactos")
        cursor.close()
        conexion.close()
def actualizar_contacto(id_contacto, nombre, telefono, email):
    conexion = conectar()
    if conexion:
        cursor = conexion.cursor()
        sql = """
        UPDATE contactos 
        SET nombre = %s, telefono = %s, email = %s 
        WHERE id = %s
        """
        cursor.execute(sql, (nombre, telefono, email, id_contacto))
        conexion.commit()
        if cursor.rowcount > 0:
            print("✏️ Contacto actualizado correctamente")
        else:
            print("⚠️ No se encontró el contacto")
        cursor.close()
        conexion.close()
def main():
    print("=== Actualizar Contacto ===")
    while True:
        print("\n1. Agregar contacto")
        print("2. Mostrar contactos")
        print("3. Actualizar contacto")
        print("4. Salir")
        opcion = input("Elige una opción: ")
        if opcion == "1":
            nombre = input("Nombre: ")
            telefono = input("Teléfono: ")
            email = input("Email: ")
            guardar_contacto(nombre, telefono, email)
        elif opcion == "2":
            mostrar_contactos()
        elif opcion == "3":
            mostrar_contactos()
            try:
                id_contacto = int(input("Ingrese el ID a actualizar: "))
                print("Ingrese los nuevos datos:")
                nombre = input("Nuevo nombre: ")
                telefono = input("Nuevo teléfono: ")
                email = input("Nuevo email: ")
                actualizar_contacto(id_contacto, nombre, telefono, email)
            except ValueError:
                print("❌ ID inválido")
        elif opcion == "4":
            print("👋 Saliendo")
            break
        else:
            print("❌ Opción inválida")
if __name__ == "__main__":
    main()

File: java/integracion_con_base_de_datos/actualizar_contacto/src/actualizar_contacto/actualizar_contacto.java
package actualizar_contacto;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.util.Scanner;
public class actualizar_contacto {
    static Scanner sc = new Scanner(System.in);
    public static void main(String[] args) {
        Connection conn = conexion.conectar();
        if (conn == null) {
            System.out.println("❌ No se pudo conectar a la base de datos");
            return;
        }
        int opcion;
        do {
            System.out.println("\n=== AGENDA DE CONTACTOS ===");
            System.out.println("1. Agregar contacto");
            System.out.println("2. Actualizar contacto");
            System.out.println("3. Salir");
            System.out.print("Seleccione una opción: ");
            opcion = sc.nextInt();
            sc.nextLine();
            switch (opcion) {
                case 1:
                    agregarContacto(conn);
                    break;
                case 2:
                    actualizarContacto(conn);
                    break;
                case 3:
                    System.out.println("👋 Saliendo del programa...");
                    break;
                default:
                    System.out.println("❌ Opción inválida");
            }
        } while (opcion != 3);
        sc.close();
    }
    public static void agregarContacto(Connection conn) {
        try {
            System.out.println("\n=== AGREGAR CONTACTO ===");
            System.out.print("Nombre: ");
            String nombre = sc.nextLine();
            System.out.print("Teléfono: ");
            String telefono = sc.nextLine();
            System.out.print("Correo: ");
            String correo = sc.nextLine();
            String sql = "INSERT INTO contactos (nombre, telefono, correo) VALUES (?, ?, ?)";
            PreparedStatement ps = conn.prepareStatement(sql);
            ps.setString(1, nombre);
            ps.setString(2, telefono);
            ps.setString(3, correo);
            int filas = ps.executeUpdate();
            if (filas > 0) {
                System.out.println("✅ Contacto agregado correctamente");
            }
        } catch (Exception e) {
            System.out.println("❌ Error al agregar: " + e.getMessage());
        }
    }
    public static void actualizarContacto(Connection conn) {
        try {
            System.out.println("\n=== ACTUALIZAR CONTACTO ===");
            System.out.print("Ingrese el ID del contacto: ");
            int id = sc.nextInt();
            sc.nextLine();
            System.out.print("Nuevo nombre: ");
            String nombre = sc.nextLine();
            System.out.print("Nuevo teléfono: ");
            String telefono = sc.nextLine();
            System.out.print("Nuevo correo: ");
            String correo = sc.nextLine();
            String sql = "UPDATE contactos SET nombre = ?, telefono = ?, correo = ? WHERE id = ?";
            PreparedStatement ps = conn.prepareStatement(sql);
            ps.setString(1, nombre);
            ps.setString(2, telefono);
            ps.setString(3, correo);
            ps.setInt(4, id);
            int filas = ps.executeUpdate();
            if (filas > 0) {
                System.out.println("✅ Contacto actualizado correctamente");
            } else {
                System.out.println("❌ No existe un contacto con ese ID");
            }
        } catch (Exception e) {
            System.out.println("❌ Error al actualizar: " + e.getMessage());
        }
    }
}